src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3101,36 +3101,44 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 	if (__level == SOL_SOCKET) {
 		switch(__optname) {
 		case SO_REUSEADDR:
+			lock_tcp_con();
```

Comment:
@OphirMunk I recommend to locate the lock() just before the update of m_pcb. After the reading the optval and logging. no point logging inside the lock.
Same comment for all the rest of the changes.